### DISCUSSION TITLE
OMPython setInputs() function

### DISCUSSION THREAD

**Discussion by FranziskaMuellr** [OpenModelica/OpenModelica#10555] [Q&A] [answered]
Hello, 
I would like to ask a question regarding the setInputs() function of OMPython. 
As an Input for my model I want to pass a (time)table/array. In OpenModelica it is working with the Combitimetable but I want to set the input in Python. In the documentations I can find an example just for setting only one value for an input but I need to pass a timeseries. Is it possible passing an input array in Python instead of using the combitimetable in Modelica? 

To demonstrate the problem I created a very simple example as follows: 
I want to set the Inputs [1,2,3,4,5] for the timesteps [1,2,3,4,5]; as an output it should give me [2,4,6,8,10] for the same timesteps.

![examplemodelica](https://user-images.githubusercontent.com/130980067/232452363-4d8370c4-f5c8-4288-9ce1-a584c3850f4d.PNG)

```
omc = OMCSessionZMQ()
mod=ModelicaSystem("TestInputsOutputs.mo", "TestInputsOutputs")
mod.buildModel()
mod.getInputs()
mod.setInputs("TestInput=[1,1;2,2;3,3;4,4;5,5]")
```

**Comment by arun3688** [comment, marked as answer]
@FranziskaMuellr  yes you can pass timeseries using `setInputs()` API in the form of array of tuples 
`[(time, value)]` , an example for your use case would be 

```
TestInput = "[(1,1),(2,2),(3, 3),(4,4),(5,5)]"
mod.setInputs(TestInput)
```
if you want to set multiple inputs

```
TestInput1 = "[(1,1),(2,2),(3, 3),(4,4),(5,5)]"
TestInput2 = "[(1,1),(2,2),(3, 3),(4,4),(5,5)]"
TestInput3 = "[(1,1),(2,2),(3, 3),(4,4),(5,5)]"

mod.setInputs(TestInput1)
mod.setInputs(TestInput2)
mod.setInputs(TestInput3)
```

**Reply by FranziskaMuellr** [reply]
Thank you. 
If I pass the TestInputs in form of array of tuples, it is accepted but after the simulation the function mod.getInputs()  still gets ‘None’ for the Inputs. 

```
omc = OMCSessionZMQ()
mod=ModelicaSystem("TestInputsOutputs.mo", "TestInputsOutputs")
mod.buildModel()
TestInput = "[(1,1),(2,2),(3, 3),(4,4),(5,5)]"
mod.setInputs({'TestInput': TestInput})
mod.simulate()
mod.getInputs()

{'TestInput': None}
```
